hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/ProtobufLogReader.java

```diff
@@ -490,6 +474,54 @@ private IOException extractHiddenEof(Exception ex) {
     return null;
   }
 
+  /**
+   * This is used to determine whether we have already reached the WALTrailer. As the size and magic
+   * are at the end of the WAL file, it is possible that these two options are missing while
+   * writing, so we will consider there is no trailer. And when we actually reach the WALTrailer, we
+   * will try to decode it as WALKey and we will fail but the error could be vary as it is parsing
```

Comment:
nit: `error could be vary` to `error could vary`. Can be fixed while committing.

Reply:
Fixed. Thanks for taking a look!